src/mpi/request/greq_start.c

```diff
@@ -266,16 +273,17 @@ int MPIX_Grequest_start(MPI_Grequest_query_function * query_fn,
     MPID_THREAD_CS_ENTER(GLOBAL, MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX);
 
     /* ... body of routine ... */
-    *request = MPI_REQUEST_NULL;
-    mpi_errno =
-        MPIX_Grequest_start_impl(query_fn, free_fn, cancel_fn, poll_fn, wait_fn, extra_state,
-                                 &lrequest_ptr);
-
-    if (mpi_errno == MPI_SUCCESS) {
-        *request = lrequest_ptr->handle;
+    MPIR_Request *request_ptr = NULL;
+    mpi_errno = MPIR_Grequest_start_impl(query_fn, free_fn, cancel_fn, extra_state, &request_ptr);
+    if (mpi_errno) {
+        goto fn_fail;
     }
+    request_ptr->u.ureq.greq_fns->poll_fn = poll_fn;
+    request_ptr->u.ureq.greq_fns->wait_fn = wait_fn;
+    *request = request_ptr->handle;
 
```

Comment:
> 
> 
> what is this change? It is not simple rename.
> 
> ```
>     request_ptr->u.ureq.greq_fns->poll_fn = poll_fn;
>     request_ptr->u.ureq.greq_fns->wait_fn = wait_fn;
> ```

`MPIX_Grequest_start` is nearly the same as `MPI_Grequest_start` except setting two additional function pointers (`poll_fn` and `wait_fn`). Rather than duplicate the code, the change now reuses `MPIR_Grequest_start_impl` (in stead of `MPIX_Grequest_start_impl`). The above two new lines are the difference that `MPIX_Grequest_start` needed.

The other reason is we have used the convention of naming MPIR impl functions as `MPIR_Xxx_impl` regardless it is for a `MPI_Xxx` function or a `MPIX_Xxx` function. Since here we have both `MPI_Grequest_start` and `MPIX_Grequest_start`. Let `MPIX_Grequest_start` call `MPIR_Grequest_start_impl` removes the need to create an exception for `MPIX_Grequest_start`.

> Besides, looks like `fn_fail` is not defined in this commit.

Thanks. Labels added.